src/server/models/listItem/listItem.ts

```diff
@@ -120,6 +120,59 @@ export async function findListItemById(id: string | number) {
   }
 }
 
+export async function findListItemByReference(ref: string): Promise<ListItem | null> {
+  try {
+    return await prisma.listItem.findUnique({
+        where: { reference: ref }
+    });
+  } catch (error) {
+    throw new Error(`findListItemByReference Error ${(error as Error).message}`);
+  }
+}
+
+export async function togglerListItemIsApproved({
```

Comment:
I made a similar comment in another PR, I think this function has been reintroduced by  merge conflict